backend/etl/processor.js

```diff
@@ -0,0 +1,215 @@
+const fetch = require('node-fetch')
+const fs = require('fs')
+
+const {
+  Discount,
+  EtlJob,
+  EtlShop,
+  Network,
+  Order,
+  Shop,
+  ShopDeployment
+} = require('../models')
+const { EtlJobStatuses, ShopDeploymentStatuses } = require('../enums')
+const { getConfig } = require('../utils/encryptedConfig')
+const { getShopDataUrl } = require('../utils/shop')
+const { DSHOP_CACHE } = require('../utils/const')
+
+const { getLogger } = require('../utils/logger')
+const log = getLogger('etl.job')
+
+const FETCH_TIMEOUT_MS = 60000 // 60 sec
+
+class EtlJobProcessor {
+  constructor(data) {
+    this.data = data
+  }
+
+  async _fetchJson(shop, filename) {
+    if (shop.published) {
```

Comment:
Are we more interested in the 'latest' data, or currently published data? Like if someone creates a shop and publishes it with zero products, then starts adding products over the course of a week or so, we'd be more interested in the unpublished data than the published data?

Reply:
We take a daily snapshot of the data. There will be one row in EtlShops per shop per day.
So we should be able to track the daily progress of a shop setup.

Before a shop is published, we track products/collections/shippings data on disk, once a shop is published we track that data by downloading it from IPFS.

Makes sense?

Reply:
Makes sense... I'm just wondering if there are downsides to always tracking the data on disk instead of IPFS.

Reply:
So once the shop is published, we always get the data from IPFS. It's only during the initial shop setup before it gets published for the first time that we would be reading the data from the disk. Does that make sense?

Reply:
It makes sense, just means we won't have visibility into unpublished changes after a shop is deployed. But maybe that's fine 🤷‍♂️

Reply:
Got it. I'll ask @gullible1 if that's something she would like. And if yes I can add it in a separate PR.